### ISSUE TITLE
[feature request] Alternating Multi-Head for Estimator-API

### ISSUE THREAD

**Issue by lhlmgr** [tensorflow/tensorflow#25722]
Hello everyone,

I really like the Estimator-API! Its clean, works well (for the most models) and it does a lot of work for you.
However, if you want to train custom models with alternating losses, its almost impossible to use the Estimator-API - unfortunate. (With tf-2.0, you definitely could use eager-mode + tfe.defun(), but I like the idea of a static graph for better performance.)

**System information**
- TensorFlow version (you are using): 1.12

**Describe the feature and the current behavior/state.**
It would be great to have a `MultiHeadSpec`, which allows to define multiple losses and train_ops as well an interval, how often the train_ops should be run.

In case of a GAN: 
- Discriminator, head1: 
  - discriminator_loss, distriminator_train_op, run train_op 2 times
- Generator, head2: 
  - generator_loss, generator_train_op, run train_op once

```python
while not mon_sess.should_stop():
   for _estimator_spec in estimator_specs:
      for _ in _estimator_spec.iterations:
         _, loss = mon_sess.run([_estimator_spec.train_op, estimator_spec.loss])
```

**Will this change the current api? How?**
No. 

**Who will benefit with this feature?**
People, who train models with alternating loss-functions.

**Any Other info.**
-

**Comment by yhliang2018** [comment]
@lhlmgr Thanks for FR! We do have [multi_head](https://github.com/tensorflow/estimator/blob/master/tensorflow_estimator/contrib/estimator/python/estimator/multi_head.py) in tf.contrib, but don't think it's suitable for GAN. If you are interested in GAN, you can take [tf-gan](https://github.com/tensorflow/tensorflow/tree/master/tensorflow/contrib/gan) as a reference.

**Comment by lhlmgr** [comment]
@yhliang2018 thanks for your reply and the references. I'm familiar with both of these tools (I used tf-gan some times, and I quickly skimmed through the `multi_head`-classes). While `multi_head` is a nice idea, it doesn't allow to alternate through the different headers. It [adds them up to a total loss](https://github.com/tensorflow/estimator/blob/master/tensorflow_estimator/contrib/estimator/python/estimator/multi_head.py#L212).
The GAN-Estimator is in my opinion a bit to inflexible (fixed learning schedule, etc.).
But, GAN was just an easy example, why a alternating `multi_head` could be useful. There are also papers like:
- [Autoencoding beyond pixels using a learned similarity metric](https://arxiv.org/abs/1512.09300)
- [Adversarial Autoencoders](https://arxiv.org/abs/1511.05644)
- .. 
which couldn't be properly trained with estimators.

Given these two mentioned libraries, I still think it would be a good idea to have such an `multi_head` that gives a bit freedom to chosen loss function / optimization schedule.

**Comment by yhliang2018** [comment]
@lhlmgr thanks for the detailed comments. Yes, I agree, the current multi_head is mainly for multi-objective learning, and not designed for GAN or other similar models. Currently this FR may not be prioritized as we mainly focus on TF 2.0 recently. Contributions on this feature are welcome to our [Addons](https://github.com/tensorflow/addons).
